### PR TITLE
feat(cli): `--verbose` and `--debug` flags

### LINKED ISSUE #4946: add --verbose and --debug flags to cli
So while trying to figure out a bug, because I'm using PnP, I tried setting them in `scripts`

```json
    "build": "DEBUG=1 graphql-codegen"
```

however this didn't work, I eventually got it working on the cli by running `DEBUG=1 yarn workspace @bb/graph-modules run graphql-codegen`, but it didn't work when I ran `yarn build` which essentially executes the above `package.json` as so 

```json
    "build": "yarn workspaces foreach --topological --verbose run build",
```

I'm not really sure why setting the env like that didn't work on OSX, from what I've read it should.

**Describe the solution you'd like**

I'd like to see `--verbose` and `--debug` added to the list of options that can be provided, this is useful over environment variables for the reason given above, but also because DEBUG and VERSION are very generic and you may have other modules installed that use those variables.

**Describe alternatives you've considered**

crying, and I tried using cross-env too, to no avail.

### PR DESCRIPTION
**Pull request by charlypoly**
Closes #4946

### CODE REVIEW

**Comment by charlypoly** [comment]
@saihaj I tested the following:

```
Charlys-MacBook-Pro~/dev/the-guild/codegen-repros/react-query-v4(master|✚3…) % yarn run codegen --verbose
yarn run v1.22.19
$ graphql-codegen --config ./codegen.yml --verbose
[STARTED] Parse Configuration
[SUCCESS] Parse Configuration
[STARTED] Generate outputs
[STARTED] Generate src/types.ts
[STARTED] Load GraphQL schemas
[SUCCESS] Load GraphQL schemas
[STARTED] Load GraphQL documents
[SUCCESS] Load GraphQL documents
[STARTED] Generate
[SUCCESS] Generate
[SUCCESS] Generate src/types.ts
[SUCCESS] Generate outputs
✨  Done in 0.92s.
```


```
Charlys-MacBook-Pro~/dev/the-guild/codegen-repros/react-query-v4(master|✚3…) % yarn run codegen --verbose --debug
yarn run v1.22.19
$ graphql-codegen --config ./codegen.yml --verbose --debug
[STARTED] Parse Configuration
[SUCCESS] Parse Configuration
[STARTED] Generate outputs
[STARTED] Generate src/types.ts
[STARTED] Load GraphQL schemas
[SUCCESS] Load GraphQL schemas
[STARTED] Load GraphQL documents
[SUCCESS] Load GraphQL documents
[STARTED] Generate
[SUCCESS] Generate
[SUCCESS] Generate src/types.ts
[SUCCESS] Generate outputs
Running lifecycle hook "afterStart" scripts...
Running lifecycle hook "afterStart" script: node test.js with args: ...
HEY!

[CLI] Loading Schemas
[CLI] Loading Documents
[CLI] Generating output
✨  Done in 0.95s.
```

```
Charlys-MacBook-Pro~/dev/the-guild/codegen-repros/react-query-v4(master|✚3…) % yarn run codegen --debug          
yarn run v1.22.19
$ graphql-codegen --config ./codegen.yml --debug
✔ Parse Configuration
✔ Generate outputs
Running lifecycle hook "afterStart" scripts...
Running lifecycle hook "afterStart" script: node test.js with args: ...
HEY!

[CLI] Loading Schemas
[CLI] Loading Documents
[CLI] Generating output
✨  Done in 1.12s.
```


The "HEY!" line comes from a custom hook node script that `console.log` to test the forwarding of logs with debug mode on.

File: packages/graphql-codegen-cli/src/utils/debugging.ts [hunk 1]
```diff
@@ -6,21 +6,17 @@ let queue: Array<{
 }> = [];
 
 export function debugLog(message: string, ...meta: any[]) {
-  if (!process.env.GQL_CODEGEN_NODEBUG && process.env.DEBUG !== undefined) {
```
Comment (saihaj) [inline packages/graphql-codegen-cli/src/utils/debugging.ts:9]: Will it still work if we do DEBUG=1 as env variable?

**Review by saihaj — APPROVED** [review]

File: packages/graphql-codegen-cli/src/utils/debugging.ts (hunk 1 above)
Comment (charlypoly) [reply packages/graphql-codegen-cli/src/utils/debugging.ts:9]: @saihaj, I'm in favor of dropping support for `DEBUG` and favor the cli flags

do you think it should be a minor then?

Comment (saihaj) [reply packages/graphql-codegen-cli/src/utils/debugging.ts:9]: I think let’s make it work with env variable and drop in next major

Comment (charlypoly) [reply packages/graphql-codegen-cli/src/utils/debugging.ts:9]: Okay, then let me refactor the PR

**Pull request merged**